src/Components/Patient/PatientHome.tsx

```diff
@@ -160,7 +130,37 @@ export const PatientHome = (props: any) => {
     },
   });
 
-  const { loading: isShiftDataLoaded, data: activeShiftingData } = useQuery(
+  const handleAssignedVolunteer = async () => {
+    const { res } = await request(routes.patchPatient, {
+      pathParams: {
+        id: patientData.id as string,
+      },
+      body: {
+        assigned_to: assignedVolunteerObject
+          ? assignedVolunteerObject.id
+          : null,
+      },
+    });
+    if (res?.ok) {
+      const dummyPatientData = Object.assign({}, patientData);
+      dummyPatientData["assigned_to"] = assignedVolunteerObject;
+      setPatientData(dummyPatientData);
```

Comment:
The patch patient API would return the updated patient object as response. Use that instead of manipulating the object from client side.

```
setPatientData(res.data);
```